src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -43,9 +42,9 @@ public BookController(BookRepository bookRepository) {
     @ApiResponses(value = {
         @ApiResponse(code = 200, message = "Successfully retrieved all books")
     })
-    public Iterable<Book> findAll() {
+    public Iterable<Book> findAll(@RequestParam(required = false) String isbn) {
 
-        return bookRepository.findAll();
+        return bookService.findAll(isbn);
```

Comment:
The idea of this card is that you had another endpoint for doing this lookup using the external API, but I think this solution is also ok :) .